packages/next/src/build/webpack/plugins/next-types-plugin/index.ts

```diff
@@ -394,6 +394,14 @@ function createRouteDefinitions() {
     dynamicRouteTypes += pluginState.routeTypes[type].dynamic
   }
 
+  // route collection is not deterministic, this makes the output of the file deterministic
+  if (staticRouteTypes) {
+    staticRouteTypes = staticRouteTypes.split("\n").sort().join("\n")
+  }
+  if (dynamicRouteTypes) {
+    dynamicRouteTypes = dynamicRouteTypes.split("\n").sort().join("\n")
+  }
```

Comment:
Seems better to make `pluginState.routeTypes` an array instead? Then it could be sorted and afterwards inserted in to the file with `.join("|")`